internal/controller/stages/stages.go

```diff
@@ -587,7 +594,10 @@ func (r *reconciler) Reconcile(
 func (r *reconciler) syncControlFlowStage(
 	ctx context.Context,
 	stage *kargoapi.Stage,
+	nowFn func() time.Time,
```

Comment:
Don't pass this in... get it from `r.nowFn` when you need it.

Reply:
Resolved in https://github.com/akuity/kargo/pull/1815/commits/fff47409b15ecc2dd8ebed60443f9b17cd8b4aae